### PR TITLE
Allows engineers access to superior welding protection and gives the Xenonauten a new welding helmet sprite.

### PR DESCRIPTION
**Pull request by Arctrooper8**
## About The Pull Request

Right now the only person who gets superior welding goggles it the CSE, a role that oftentimes simply does not show up in the round at all. Therefore giving squad engineers the ability to purchase superior welding goggles, and a new superior welding module, allows this interesting item to actually be used. 

Normal welding goggles:
00normal goggles
Superior welding goggles:
00 superior goggles
00SuperiorWelding
(Price subject to change)

That's not all though, alongside the new module comes new Xenonauten sprites for both superior and standard.
Standard:
00 helmet welding
Standard Active:
00 helmet welding on
Superior:
00 superior helmet welding
Superior Active:
00 superior helmet welding on
Why does it glow? Because it's cool and we're in the future with teleporters and laser guns. Rule of cool dude.

One final change, with Xenonauten and Jaeger now having three different helmet module sprites, helmet modules now come in crates just like their suit counterparts.
Welding:
00 welding crate
Binoculars:
00 binoculars crate
Tyr:
00 tyr crate
Antenna:
00 antenna crate
Mimir:
00 mimir crate
Some of them are just shrunken bigger crates while others I have made from scratch.

## Why It's Good For The Game

Right now superior welding protection is limited only to the CSE, a role that barely shows up. And CSE is a shipside role anyways that can rarely deploy to the fob, they're not exactly getting the most use out of it. By giving it to squad engineers, for a price, it can be made use of by those who would benefit the most from it. 
New welding helmet sprite is good to further separate Xenonauten and Jaeger visually and allows xenos to more easily read that a marine is using a welding module (Current welding one looks similar to the binocs module).
And just as the suit module turned into crates so too shall the helmet modules follow, there are a whole three modules that have different Xenonauten and Jaeger sprites, it's silly to have them all have a ground sprite that looks like Jaeger.

## Changelog
:cl:
add: Xenonauten welding helmet module now has a new sprite.
add: Helmet modules now come in miniature crates similar to their suit module counterparts.
balance: gives squad engineers access to superior welding protection.
code: added var/base_icon in the attachments file, an alternative for initial_icon.
/:cl:

### CODE REVIEW

**Review by Kuro020 — APPROVED** [review]

File: code/modules/clothing/modular_armor/attachments.dm [hunk 1]
```diff
@@ -40,6 +40,9 @@
 	///Light modifier for attachment to an armor piece
 	var/light_mod = 0
 
+	///Replacement for initial icon that allows for the code to work with multiple variants
+	var/base_icon
```
Comment (TiviPlus) [inline code/modules/clothing/modular_armor/attachments.dm:44]: Why did you change this to not use inital() anymore?

File: code/modules/clothing/modular_armor/attachments/modules.dm [hunk 2]
```diff
@@ -244,11 +246,27 @@
...
+	slot = ATTACHMENT_SLOT_HEAD_MODULE
+	variants_by_parent_type = list(/obj/item/clothing/head/modular/marine/m10x = "welding_head_superior_xn", /obj/item/clothing/head/modular/marine/m10x/leader = "welding_head_superior_xn")
+	flags_attach_features = ATTACH_REMOVABLE|ATTACH_ACTIVATION|ATTACH_APPLY_ON_MOB
+	active = FALSE
+	prefered_slot = SLOT_HEAD
+
+/obj/item/armor_module/module/welding/superior/on_attach(obj/item/attaching_to, mob/user)
+	. = ..()
+	parent.AddComponent(/datum/component/clothing_tint, TINT_4, active)
+	base_icon = icon_state
```
Comment (TiviPlus) [inline code/modules/clothing/modular_armor/attachments/modules.dm:269]: Tint component attached twice

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/clothing/modular_armor/attachments.dm (hunk 1 above)
Comment (Arctrooper8) [reply code/modules/clothing/modular_armor/attachments.dm:44]: Hyper helped me and we found that the only way to have a swapping module that could use both the Xenonauten sprite and Jaeger sprite was to make a new var. Because initial icon uses the sprite on the code (which is the Jaeger sprite) the parent variant would set it to Xenonauten but activating the welding module would then set it to Jaeger because of how the attachment sprite code worked. Base value could be set to the parent variant which would save the Xenonauten sprite for the sprite changing stuff. Now I probably didn't describe it very well, and I could try showing you over discord if you want.
TLDR: The code that handles sprite change for active and inactive modes needs base icon to save the Xenonauten sprite and not revert to Jaeger like Initial icon does, I can explain it better on discord if you'd like.

File: code/modules/clothing/modular_armor/attachments/modules.dm (hunk 2 above)
Comment (Arctrooper8) [reply code/modules/clothing/modular_armor/attachments/modules.dm:269]: Is there a way you can change the tint strength without attaching the whole component?

Comment (TiviPlus) [reply code/modules/clothing/modular_armor/attachments/modules.dm:269]: make it a passed var or a signal to update

File: code/modules/clothing/modular_armor/attachments/modules.dm [hunk 3]
```diff
@@ -222,6 +223,7 @@
 /obj/item/armor_module/module/welding/on_attach(obj/item/attaching_to, mob/user)
 	. = ..()
 	parent.AddComponent(/datum/component/clothing_tint, TINT_5, active)
+	base_icon = icon_state
```
Comment (hyper2snyper) [inline code/modules/clothing/modular_armor/attachments/modules.dm:226]: This should be set in the parent proc.

**Review by hyper2snyper — CHANGES_REQUESTED** [review]

File: code/modules/clothing/modular_armor/attachments/modules.dm [hunk 4]
```diff
@@ -244,11 +246,27 @@
...
+	item_state = "welding_head_a"
+	slot = ATTACHMENT_SLOT_HEAD_MODULE
+	variants_by_parent_type = list(/obj/item/clothing/head/modular/marine/m10x = "welding_head_superior_xn", /obj/item/clothing/head/modular/marine/m10x/leader = "welding_head_superior_xn")
+	flags_attach_features = ATTACH_REMOVABLE|ATTACH_ACTIVATION|ATTACH_APPLY_ON_MOB
+	active = FALSE
+	prefered_slot = SLOT_HEAD
+
+/obj/item/armor_module/module/welding/superior/on_attach(obj/item/attaching_to, mob/user)
+	. = ..()
+	parent.AddComponent(/datum/component/clothing_tint, TINT_4, active)
```
[the remark this answers is not in the document]
Comment (Arctrooper8) [reply code/modules/clothing/modular_armor/attachments/modules.dm:268]: Tivi said to leave it like this

File: code/modules/clothing/modular_armor/attachments/modules.dm (hunk 3 above)
Comment (Arctrooper8) [reply code/modules/clothing/modular_armor/attachments/modules.dm]: Fixed

File: code/modules/clothing/modular_armor/attachments.dm [hunk 5]
```diff
@@ -67,8 +70,10 @@
 	if(CHECK_BITFIELD(flags_attach_features, ATTACH_ACTIVATION))
 		RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, .proc/handle_actions)
 	if(!length(variants_by_parent_type) || !(parent.type in variants_by_parent_type))
+		base_icon = icon_state
 		return
 	icon_state = variants_by_parent_type[parent.type]
```
Comment (hyper2snyper) [inline code/modules/clothing/modular_armor/attachments.dm:75]:
```suggestion
	base_icon = variants_by_parent_type[parent.type]
```
and delete the line below this

File: code/modules/clothing/modular_armor/attachments.dm [hunk 6]
```diff
@@ -196,7 +201,7 @@
 
 /obj/item/armor_module/armor/Initialize()
 	. = ..()
-	item_state = initial(icon_state) + "_a"
+	item_state = base_icon + "_a"
```
Comment (hyper2snyper) [inline code/modules/clothing/modular_armor/attachments.dm:204]: i dont believe `item_state` does anything here. just delete this line.
(test though because it could just be me forgetting stuff)

**Review by hyper2snyper — CHANGES_REQUESTED** [review]
and then we are good for a TM/Merge

File: code/modules/clothing/modular_armor/attachments.dm (hunk 6 above)
Comment (Arctrooper8) [reply code/modules/clothing/modular_armor/attachments.dm:204]: done and tested

File: code/modules/clothing/modular_armor/attachments.dm (hunk 5 above)
Comment (Arctrooper8) [reply code/modules/clothing/modular_armor/attachments.dm:75]: discussed in discord

**Review by TiviPlus — APPROVED** [review]
